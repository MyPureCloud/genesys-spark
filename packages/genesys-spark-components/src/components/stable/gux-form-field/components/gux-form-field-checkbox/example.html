<h1>gux-form-field-checkbox</h1>

<p>
  slot a input[type=checkbox] element into a gux-form-field-checkbox element
</p>
<p>
  Recommended to use within the `gux-form-field-checkbox-group-beta` component.
</p>

<form onchange="notify(event)" oninput="notify(event)">
  <h2>Example 1 - Default</h2>
  <gux-form-field-checkbox>
    <input slot="input" type="checkbox" name="food-1[]" value="pizza" />
    <label slot="label">Pizza</label>
  </gux-form-field-checkbox>
  <gux-button-slot style="margin-block-start: 10px">
    <button
      type="button"
      onclick="(function () {
        const field = document.querySelector(`input[name='food-1[]'][value='pizza']`)
        field.checked = !field.checked;
      })()"
    >
      Toggle Pizza field
    </button>
  </gux-button-slot>
  <h2>Example 2 - Initial Checked State</h2>
  <gux-form-field-checkbox>
    <input slot="input" type="checkbox" name="food-1[]" value="pasta" checked />
    <label slot="label">Pasta</label>
  </gux-form-field-checkbox>
  <h2>Example 3 - Indeterminate State</h2>
  <gux-form-field-checkbox>
    <input slot="input" type="checkbox" name="food-1[]" value="hamburger" />
    <label slot="label">Hamburger</label>
  </gux-form-field-checkbox>

  <gux-button-slot style="margin-block-start: 10px">
    <button
      type="button"
      onclick="(function () {
        document.querySelector(`input[name='food-1[]'][value='hamburger']`).indeterminate = true;
      })()"
    >
      Set Hamburger field to indeterminate
    </button>
  </gux-button-slot>
  <h2>Example 4 - Disabled</h2>
  <gux-form-field-checkbox>
    <input
      slot="input"
      type="checkbox"
      name="food-1[]"
      value="sandwich"
      disabled
    />
    <label slot="label">Sandwich</label>
  </gux-form-field-checkbox>
  <h2>Example 5 - Error Message</h2>
  <gux-form-field-checkbox>
    <input slot="input" type="checkbox" name="food-1[]" value="sushi" />
    <label slot="label">Sushi</label>
    <span slot="error">Subject to availability</span>
  </gux-form-field-checkbox>
  <h2>Example 6 - Help Text</h2>
  <gux-form-field-checkbox>
    <input slot="input" type="checkbox" name="food-1[]" value="spaghetti" />
    <label slot="label">Spaghetti</label>
    <span slot="help">This is a help message</span>
  </gux-form-field-checkbox>
</form>
