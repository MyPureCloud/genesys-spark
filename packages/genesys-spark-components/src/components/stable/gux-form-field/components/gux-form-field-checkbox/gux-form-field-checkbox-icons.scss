@use '~genesys-spark/dist/scss/checkbox-icons.scss';

@mixin unchecked {
  @include checkbox-icons.checkbox-unchecked;

  background: var(--gse-ui-checkbox-icon-default-unselectedForegroundColor);
}

@mixin unchecked-hover {
  @include checkbox-icons.checkbox-unchecked-hover;

  background: var(--gse-ui-checkbox-icon-hover-foregroundColor);
}

@mixin checked {
  @include checkbox-icons.checkbox-checked;

  background: var(--gse-ui-checkbox-icon-default-selectedForegroundColor);
}

@mixin checked-hover {
  @include checkbox-icons.checkbox-checked-hover;

  background: var(--gse-ui-checkbox-icon-hover-foregroundColor);
}

@mixin unchecked-error {
  @include checkbox-icons.checkbox-unchecked-error;

  background: var(--gse-ui-checkbox-icon-error-foregroundColor);
}

@mixin checked-error {
  @include checkbox-icons.checkbox-checked-error;

  background: var(--gse-ui-checkbox-icon-error-foregroundColor);
}

@mixin indeterminate {
  @include checkbox-icons.checkbox-indeterminate;

  background: var(--gse-ui-checkbox-icon-default-selectedForegroundColor);
}

@mixin indeterminate-hover {
  @include checkbox-icons.checkbox-indeterminate-hover;

  background: var(--gse-ui-checkbox-icon-hover-foregroundColor);
}

@mixin indeterminate-error {
  @include checkbox-icons.checkbox-indeterminate-error;

  background: var(--gse-ui-checkbox-icon-error-foregroundColor);
}
