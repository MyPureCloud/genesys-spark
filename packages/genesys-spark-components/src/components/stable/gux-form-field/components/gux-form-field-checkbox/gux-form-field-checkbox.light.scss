// This file is needed because of a safari
//  - https://bugs.webkit.org/show_bug.cgi?id=223814
//  - https://bugs.webkit.org/show_bug.cgi?id=178237#attach_442680

@use './gux-form-field-checkbox-icons.scss';

gux-form-field-checkbox {
  input[type='checkbox']::before {
    grid-area: 1/1;
    content: '';
    border-radius: 15%;
  }

  input[type='checkbox']:focus-visible {
    border-radius: var(--gse-ui-checkbox-focus-borderRadiusSmall);
    outline: var(--gse-ui-checkbox-focus-border-width)
      var(--gse-ui-checkbox-focus-border-style)
      var(--gse-ui-checkbox-focus-border-color);
    outline-offset: var(--gse-ui-checkbox-focus-offset);
  }

  input[type='checkbox']:not(:checked)::before {
    @include gux-form-field-checkbox-icons.unchecked;
  }

  input[type='checkbox']:not(:checked):not(:disabled):not(
      :indeterminate
    ):hover::before {
    @include gux-form-field-checkbox-icons.unchecked-hover;
  }

  input[type='checkbox']:checked:not(:disabled):hover::before {
    @include gux-form-field-checkbox-icons.checked-hover;
  }

  input[type='checkbox']:checked::before {
    @include gux-form-field-checkbox-icons.checked;
  }

  input[type='checkbox']:indeterminate::before {
    @include gux-form-field-checkbox-icons.indeterminate;
  }

  input[type='checkbox']:not(:disabled):indeterminate:hover {
    @include gux-form-field-checkbox-icons.indeterminate-hover;
  }

  input[type='checkbox']:disabled::before {
    cursor: not-allowed;
    opacity: var(--gse-ui-checkbox-disabled-opacity);
  }

  &.gux-input-error {
    input[type='checkbox']:not(:checked)::before {
      @include gux-form-field-checkbox-icons.unchecked-error;
    }

    input[type='checkbox']:checked::before {
      @include gux-form-field-checkbox-icons.checked-error;
    }

    input[type='checkbox']:indeterminate::before {
      @include gux-form-field-checkbox-icons.indeterminate-error;
    }
  }
}
