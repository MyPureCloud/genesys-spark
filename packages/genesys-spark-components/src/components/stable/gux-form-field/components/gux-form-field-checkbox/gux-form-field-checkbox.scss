@use '../../functional-components/gux-form-field-error/gux-form-field-error.scss';
@use '../../functional-components/gux-form-field-help/gux-form-field-help.scss';
@use '~genesys-spark/dist/scss/mixins.scss';
@use '~genesys-spark/dist/scss/checkbox-icons.scss';

@include gux-form-field-error.Style;
@include gux-form-field-help.Style;

:host {
  display: block;
}

:host(.gux-input-error) {
  ::slotted(input[type='checkbox']:not(:checked))::before {
    @include checkbox-icons.unchecked-error;

    background: var(--gse-ui-checkbox-icon-error-foregroundColor);
  }

  ::slotted(input[type='checkbox']:checked)::before {
    @include checkbox-icons.checked-error;

    background: var(--gse-ui-checkbox-icon-error-foregroundColor);
  }

  ::slotted(
      input[type='checkbox']:not(:checked):not(:disabled):not(
          :indeterminate
        ):hover
    )::before {
    @include checkbox-icons.unchecked-error;

    background: var(--gse-ui-checkbox-icon-error-foregroundColor);
  }

  ::slotted(input[type='checkbox']:indeterminate)::before {
    @include checkbox-icons.indeterminate-error;

    background: var(--gse-ui-checkbox-icon-error-foregroundColor);
  }
}

:host(.gux-disabled) {
  cursor: not-allowed;
  opacity: var(--gse-ui-checkbox-disabled-opacity);

  ::slotted(label) {
    cursor: not-allowed;
  }
}

.gux-input-label {
  display: inline-flex;
  flex-direction: row;
  gap: var(--gse-ui-checkbox-gap);

  .gux-label-screenreader {
    @include mixins.gux-sr-only;
  }
}

::slotted(input[type='checkbox']) {
  display: inline-grid;
  width: var(--gse-ui-checkbox-icon-width);
  height: var(--gse-ui-checkbox-icon-height);
  margin: 0;
  text-align: center;
  vertical-align: middle;
  -webkit-appearance: none;
  appearance: none;
  cursor: pointer;
  border: 0;
  outline: none;
}

::slotted(input[type='checkbox'])::before {
  grid-area: 1/1;
  content: '';
  border-radius: 15%;
}

::slotted(input[type='checkbox']:focus-visible) {
  border-radius: var(--gse-ui-checkbox-focus-borderRadiusSmall);
  outline: var(--gse-ui-checkbox-focus-border-width)
    var(--gse-ui-checkbox-focus-border-style)
    var(--gse-ui-checkbox-focus-border-color);
  outline-offset: var(--gse-ui-checkbox-focus-offset);
}

::slotted(input[type='checkbox']:not(:checked))::before {
  @include checkbox-icons.unchecked;

  background: var(--gse-ui-checkbox-icon-default-unselectedForegroundColor);
}

::slotted(
    input[type='checkbox']:not(:checked):not(:disabled):not(
        :indeterminate
      ):hover
  )::before {
  @include checkbox-icons.unchecked-hover;

  background: var(--gse-ui-checkbox-icon-hover-foregroundColor);
}

::slotted(input[type='checkbox']:checked:not(:disabled):hover) {
  @include checkbox-icons.checked-hover;

  background: var(--gse-ui-checkbox-icon-hover-foregroundColor);
}

::slotted(input[type='checkbox']:not(:disabled):indeterminate:hover) {
  @include checkbox-icons.indeterminate-hover;

  background: var(--gse-ui-checkbox-icon-hover-foregroundColor);
}

::slotted(input[type='checkbox']:checked)::before {
  @include checkbox-icons.checked;

  background: var(--gse-ui-checkbox-icon-default-selectedForegroundColor);
}

::slotted(input[type='checkbox']:indeterminate)::before {
  @include checkbox-icons.indeterminate;

  background: var(--gse-ui-checkbox-icon-default-selectedForegroundColor);
}

::slotted(input[type='checkbox']:disabled)::before {
  cursor: not-allowed;
  opacity: var(--gse-ui-checkbox-disabled-opacity);
}

::slotted(label) {
  display: inline-block;
  font-family: var(--gse-ui-checkbox-label-text-fontFamily);
  font-size: 12px; // -- missing font-size token https://inindca.atlassian.net/browse/COMUI-1672
  font-weight: var(--gse-ui-checkbox-label-text-fontWeight);
  line-height: var(--gse-ui-checkbox-label-text-lineHeight);
  color: var(--gse-ui-checkbox-label-foregroundColor);
  vertical-align: middle;
}
