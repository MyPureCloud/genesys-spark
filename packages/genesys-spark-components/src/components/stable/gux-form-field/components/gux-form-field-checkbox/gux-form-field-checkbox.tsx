import { Component, Element, h, Host, JSX, State, Prop } from '@stencil/core';

import { calculateInputDisabledState } from '@utils/dom/calculate-input-disabled-state';
import { onInputDisabledStateChange } from '@utils/dom/on-input-disabled-state-change';
import { OnMutation } from '@utils/decorator/on-mutation';
import { preventBrowserValidationStyling } from '@utils/dom/prevent-browser-validation-styling';
import { hasSlot } from '@utils/dom/has-slot';

import {
  GuxFormFieldError,
  GuxFormFieldHelp
} from '../../functional-components/functional-components';

import { trackComponent } from '@utils/tracking/usage';
import { validateFormIds, getSlottedInput } from '../../gux-form-field.service';

/**
 * @slot input - Required slot for input tag
 * @slot label - Required slot for label tag
 * @slot error - Optional slot for error message
 * @slot help - Optional slot for help message
 */
@Component({
  styleUrl: 'gux-form-field-checkbox.scss',
  tag: 'gux-form-field-checkbox',
  shadow: true
})
export class GuxFormFieldCheckbox {
  private input: HTMLInputElement;
  private disabledObserver: MutationObserver;

  @Element()
  private root: HTMLElement;

  @Prop()
  labelPosition: 'beside' | 'screenreader' = 'beside';

  @State()
  private disabled: boolean;

  @State()
  private hasError: boolean = false;

  @State()
  private hasHelp: boolean = false;

  @OnMutation({ childList: true, subtree: true })
  onMutation(): void {
    this.hasError = hasSlot(this.root, 'error');
    this.hasHelp = hasSlot(this.root, 'help');
  }

  componentWillLoad(): void {
    this.setInput();

    this.hasError = hasSlot(this.root, 'error');
    this.hasHelp = hasSlot(this.root, 'help');

    trackComponent(this.root, { variant: this.variant });
  }

  disconnectedCallback(): void {
    if (this.disabledObserver) {
      this.disabledObserver.disconnect();
    }
  }

  render(): JSX.Element {
    return (
      <Host
        class={{
          'gux-input-error': this.hasError,
          'gux-disabled': this.disabled
        }}
      >
        <div class="gux-form-field-container">
          <div class="gux-input-label">
            <div class="gux-input">
              <slot name="input" onSlotchange={() => this.setInput()} />
            </div>
            <div class={`gux-label-${this.labelPosition}`}>
              <slot name="label" />
              <GuxFormFieldError show={this.hasError}>
                <slot name="error" />
              </GuxFormFieldError>
              <GuxFormFieldHelp show={!this.hasError && this.hasHelp}>
                <slot name="help" />
              </GuxFormFieldHelp>
            </div>
          </div>
        </div>
      </Host>
    ) as JSX.Element;
  }

  private get variant(): string {
    return this.labelPosition.toLowerCase();
  }

  private setInput(): void {
    this.input = getSlottedInput(
      this.root,
      'input[type="checkbox"][slot="input"]'
    );

    preventBrowserValidationStyling(this.input);

    this.disabled = calculateInputDisabledState(this.input);

    this.disabledObserver = onInputDisabledStateChange(
      this.input,
      (disabled: boolean) => {
        this.disabled = disabled;
      }
    );

    validateFormIds(this.root, this.input);
  }
}
