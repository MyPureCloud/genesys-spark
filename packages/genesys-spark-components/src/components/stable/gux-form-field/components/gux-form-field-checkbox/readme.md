# gux-form-field-checkbox



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type                         | Default    |
| --------------- | ---------------- | ----------- | ---------------------------- | ---------- |
| `labelPosition` | `label-position` |             | `"beside" \| "screenreader"` | `'beside'` |


## Slots

| Slot      | Description                     |
| --------- | ------------------------------- |
| `"error"` | Optional slot for error message |
| `"help"`  | Optional slot for help message  |
| `"input"` | Required slot for input tag     |
| `"label"` | Required slot for label tag     |


## Dependencies

### Used by

 - [gux-all-row-select](../../../gux-table/gux-all-row-select)
 - [gux-column-manager](../../../gux-column-manager)
 - [gux-row-select](../../../gux-table/gux-row-select)

### Depends on

- [gux-icon](../../../gux-icon)

### Graph
```mermaid
graph TD;
  gux-form-field-checkbox --> gux-icon
  gux-all-row-select --> gux-form-field-checkbox
  gux-column-manager --> gux-form-field-checkbox
  gux-row-select --> gux-form-field-checkbox
  style gux-form-field-checkbox fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
