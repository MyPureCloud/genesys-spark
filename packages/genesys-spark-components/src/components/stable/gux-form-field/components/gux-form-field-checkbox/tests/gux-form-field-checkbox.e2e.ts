import { E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../test/e2eTestUtils';

import { renderConfigs } from './gux-form-field-checkbox.common';

const axeExclusions = [];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-form-field-checkbox', () => {
  describe('#render', () => {
    renderConfigs.forEach(({ html }, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const snapshotPage = await newNonrandomE2EPage({ html });
        const element = await snapshotPage.find('gux-form-field-checkbox');
        const elementShadowDom = await element.find(
          'pierce/.gux-form-field-container'
        );

        expect(element.outerHTML).toMatchSnapshot();
        expect(elementShadowDom).toMatchSnapshot();

        const accessibilityPage = await newSparkE2EPage({ html });

        await a11yCheck(accessibilityPage, axeExclusions);
      });
    });
  });

  it('switches between states when clicked', async () => {
    const page = await newSparkE2EPage({
      html: `
        <gux-form-field-checkbox>
          <input slot="input" type="checkbox" name="food-1[]" value="pizza"/>
          <label slot="label">Pizza</label>
        </gux-form-field-checkbox>
      `
    });

    const element = await page.find('gux-form-field-checkbox');
    const label = await element.find('label');
    const input = await element.find('input');

    await a11yCheck(page, axeExclusions, 'Before checking checkbox');
    await label.click();
    await page.waitForChanges();
    expect(await input.getProperty('checked')).toBe(true);

    await a11yCheck(page, axeExclusions, 'After checking checkbox');
    await label.click();
    await page.waitForChanges();
    expect(await input.getProperty('checked')).toBe(false);

    await label.click();
    await page.waitForChanges();
    expect(await input.getProperty('checked')).toBe(true);
  });
});
