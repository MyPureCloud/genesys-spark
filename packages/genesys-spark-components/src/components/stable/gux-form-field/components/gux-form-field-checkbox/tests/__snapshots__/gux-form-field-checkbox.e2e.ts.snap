// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-checkbox #render should render component as expected (1) 1`] = `"<gux-form-field-checkbox hydrated=\\"\\"> <input slot=\\"input\\" type=\\"checkbox\\" name=\\"food-1[]\\" value=\\"pizza\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Pizza</label> </gux-form-field-checkbox>"`;

exports[`gux-form-field-checkbox #render should render component as expected (1) 2`] = `
<div class="gux-form-field-container">
  <div class="gux-input-label">
    <div class="gux-input">
      <slot name="input"></slot>
    </div>
    <div class="gux-label-beside">
      <slot name="label"></slot>
      <div class="gux-form-field-error">
        <gux-icon hydrated="" icon-name="alert-warning-octogon" size="inherit"></gux-icon>
        <div class="gux-message">
          <slot name="error"></slot>
        </div>
      </div>
      <div class="gux-form-field-help">
        <div class="gux-message">
          <slot name="help"></slot>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-checkbox #render should render component as expected (2) 1`] = `"<gux-form-field-checkbox class=\\"gux-disabled\\" hydrated=\\"\\"> <input slot=\\"input\\" type=\\"checkbox\\" name=\\"food-1[]\\" value=\\"pizza\\" disabled=\\"\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Pizza</label> </gux-form-field-checkbox>"`;

exports[`gux-form-field-checkbox #render should render component as expected (2) 2`] = `
<div class="gux-form-field-container">
  <div class="gux-input-label">
    <div class="gux-input">
      <slot name="input"></slot>
    </div>
    <div class="gux-label-beside">
      <slot name="label"></slot>
      <div class="gux-form-field-error">
        <gux-icon hydrated="" icon-name="alert-warning-octogon" size="inherit"></gux-icon>
        <div class="gux-message">
          <slot name="error"></slot>
        </div>
      </div>
      <div class="gux-form-field-help">
        <div class="gux-message">
          <slot name="help"></slot>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-checkbox #render should render component as expected (3) 1`] = `"<gux-form-field-checkbox class=\\"gux-input-error\\" hydrated=\\"\\"> <input slot=\\"input\\" type=\\"checkbox\\" name=\\"food-1[]\\" value=\\"pizza\\" id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Pizza</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">This is an error message</span> </gux-form-field-checkbox>"`;

exports[`gux-form-field-checkbox #render should render component as expected (3) 2`] = `
<div class="gux-form-field-container">
  <div class="gux-input-label">
    <div class="gux-input">
      <slot name="input"></slot>
    </div>
    <div class="gux-label-beside">
      <slot name="label"></slot>
      <div class="gux-form-field-error gux-show">
        <gux-icon hydrated="" icon-name="alert-warning-octogon" size="inherit"></gux-icon>
        <div class="gux-message">
          <slot name="error"></slot>
        </div>
      </div>
      <div class="gux-form-field-help">
        <div class="gux-message">
          <slot name="help"></slot>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-checkbox #render should render component as expected (4) 1`] = `"<gux-form-field-checkbox hydrated=\\"\\"> <input slot=\\"input\\" type=\\"checkbox\\" name=\\"food-1[]\\" value=\\"pizza\\" checked=\\"\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Pizza</label> </gux-form-field-checkbox>"`;

exports[`gux-form-field-checkbox #render should render component as expected (4) 2`] = `
<div class="gux-form-field-container">
  <div class="gux-input-label">
    <div class="gux-input">
      <slot name="input"></slot>
    </div>
    <div class="gux-label-beside">
      <slot name="label"></slot>
      <div class="gux-form-field-error">
        <gux-icon hydrated="" icon-name="alert-warning-octogon" size="inherit"></gux-icon>
        <div class="gux-message">
          <slot name="error"></slot>
        </div>
      </div>
      <div class="gux-form-field-help">
        <div class="gux-message">
          <slot name="help"></slot>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-checkbox #render should render component as expected (5) 1`] = `"<gux-form-field-checkbox class=\\"gux-input-error\\" hydrated=\\"\\"> <input slot=\\"input\\" type=\\"checkbox\\" name=\\"food-1[]\\" value=\\"pizza\\" id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-error-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Pizza</label> <span slot=\\"error\\" id=\\"gux-form-field-error-i\\">Error message</span> </gux-form-field-checkbox>"`;

exports[`gux-form-field-checkbox #render should render component as expected (5) 2`] = `
<div class="gux-form-field-container">
  <div class="gux-input-label">
    <div class="gux-input">
      <slot name="input"></slot>
    </div>
    <div class="gux-label-beside">
      <slot name="label"></slot>
      <div class="gux-form-field-error gux-show">
        <gux-icon hydrated="" icon-name="alert-warning-octogon" size="inherit"></gux-icon>
        <div class="gux-message">
          <slot name="error"></slot>
        </div>
      </div>
      <div class="gux-form-field-help">
        <div class="gux-message">
          <slot name="help"></slot>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-checkbox #render should render component as expected (6) 1`] = `"<gux-form-field-checkbox hydrated=\\"\\"> <input slot=\\"input\\" type=\\"checkbox\\" name=\\"food-1[]\\" value=\\"pizza\\" id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-help-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Pizza</label> <span slot=\\"help\\" id=\\"gux-form-field-help-i\\">This is a help message</span> </gux-form-field-checkbox>"`;

exports[`gux-form-field-checkbox #render should render component as expected (6) 2`] = `
<div class="gux-form-field-container">
  <div class="gux-input-label">
    <div class="gux-input">
      <slot name="input"></slot>
    </div>
    <div class="gux-label-beside">
      <slot name="label"></slot>
      <div class="gux-form-field-error">
        <gux-icon hydrated="" icon-name="alert-warning-octogon" size="inherit"></gux-icon>
        <div class="gux-message">
          <slot name="error"></slot>
        </div>
      </div>
      <div class="gux-form-field-help gux-show">
        <div class="gux-message">
          <slot name="help"></slot>
        </div>
      </div>
    </div>
  </div>
</div>
`;
